Adds monitor logger extension

cpeeyush opened the pull request:
- Adds a monitor logger extension.
- java.util.logging is used for logger. This will not introduce any additional dependency.
- Adds unit tests.
- Adds documentation.

Closes #297 (https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/297)

jimmarino commented:
Hi @cpeeyush Thanks for the contribution. Two minor things, can you jdk-logger-monitor to indicate which impl it uses and also get it to pass the Codestyle checks?

Thanks

cpeeyush commented:
> Hi @cpeeyush Thanks for the contribution. Two minor things, can you rename the module to jdk-logger-monitor to indicate which impl it uses and also get it to pass the Codestyle checks?
> 
> Thanks

Hi @jimmarino,
Thank for your feedback. I have renamed module to jdk-logger-monitor and also fixed the checkstyle warnings. Please take a look now. Thank you.
